src/software/thunderscope/gl/layers/gl_referee_info_layer.py

```diff
@@ -0,0 +1,250 @@
+    # outline color of the tolerance region for ball placement (when the ball is inside)
+    BALL_PLACEMENT_TOLERANCE_VISUALIZATION_COLOR_ON = Colors.GREEN
+    # outline color of the tolerance region for ball placement (when the ball is outside)
+    BALL_PLACEMENT_TOLERANCE_VISUALIZATION_COLOR_OFF = Colors.RED
+    # outline color of the target mark
+    BALL_PLACEMENT_TARGET_VISUALIZATION_COLOR = Colors.FIELD_LINE_COLOR
+    # text color for count down
+    COUNT_DOWN_TEXT_COLOR = Colors.RED
+
+    def __init__(self, name: str, buffer_size: int = 1) -> None:
+        """Initialize the GLRefereeInfoLayer
+
+        :param name: The displayed name of the layer
+        :param buffer_size: the buffer size, set higher for smoother plots.
+                            Set lower for more realtime plots. Default is arbitrary.
+        """
+        super().__init__(name)
+        self.setDepthValue(DepthValues.OVERLAY_DEPTH)
+        self.referee_vis_buffer = ThreadSafeBuffer(buffer_size, Referee)
+        self.ball_placement_vis_buffer = ThreadSafeBuffer(
+            buffer_size, BallPlacementVisualization
+        )
+        self.world_buffer = ThreadSafeBuffer(buffer_size, World)
+
+        self.cached_world = None
+        self.cached_referee_info = None
+
+        self.referee_text_graphics = ObservableList(self._graphics_changed)
+
+        self.placement_tolerance_graphic = GLCircle(
+            parent_item=self,
+            radius=BALL_PLACEMENT_TOLERANCE_RADIUS_METERS,
+            outline_color=self.BALL_PLACEMENT_TOLERANCE_VISUALIZATION_COLOR_OFF,
+        )
+
+        self.placement_target_graphic = GLCircle(
+            parent_item=self,
+            radius=BALL_PLACEMENT_TOLERANCE_RADIUS_METERS,
+            outline_color=self.BALL_PLACEMENT_TARGET_VISUALIZATION_COLOR,
+        )
+
+        self.robot_avoid_circle_graphic = GLCircle(
+            parent_item=self,
+            radius=BALL_PLACEMENT_ROBOT_AVOID_RADIUS_METERS,
+            outline_color=self.BALL_PLACEMENT_ROBOT_AVOID_AREA_VISUALIZATION_COLOR,
+        )
+
+        # initialize the two text items to display
+        self.gamestate_type_text: Optional[GLLabel] = None
+        self.command_type_text: Optional[GLLabel] = None
+
+        self.ball_placement_countdown_graphic: Optional[GLTextItem] = None
+
+        self.ball_placement_point = None
+        self.ball_placement_in_progress = False
+        self.ball_placement_tolerance_circle = None
+        self.shrink_target = True
+        self.placement_start_time = 0
+
+    def __update_ball_placement(self) -> None:
+        """Update ball placement visuals"""
+        ball_placement_vis_proto = self.ball_placement_vis_buffer.get(
+            block=False, return_cached=False
+        )
+
+        if not self.ball_placement_countdown_graphic:
+            self.ball_placement_countdown_graphic = GLTextItem(
+                parentItem=self,
+                font=QtGui.QFont("Roboto", 7, weight=700),
```

Comment:
It seems like we've standardized "Roboto" as our font of choice. Can you abstract it into a constant in `constants.py` and replace all uses?

Reply:
Done